src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1191,7 +1192,13 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
             // * Otherwise, the argument is Type.Missing.
 
             BoundExpression? defaultValue = null;
-            if (parameter.IsMarshalAsObject)
+            if (InAttributeArgument)
+            {
+                // CS7067: Attribute constructor parameter '{0}' is optional, but no default parameter value was specified.
+                diagnostics.Add(ErrorCode.ERR_BadAttributeParamDefaultArgument, syntax.Location, parameter.Name);
+                defaultValue = BadExpression(syntax).MakeCompilerGenerated();
```

Comment:
>defaultValue = BadExpression(syntax).MakeCompilerGenerated(); [](http://example.com/codeflow?start=16&length=61)

Consider removing since the `return` expression handles this case.